freeipa/src/main/java/com/sequenceiq/freeipa/service/freeipa/user/CloudIdentitySyncService.java

```diff
@@ -48,16 +58,59 @@ private void syncAzureObjectIds(Stack stack, UmsUsersState umsUsersState, BiCons
 
             SetRangerCloudIdentityMappingRequest setRangerCloudIdentityMappingRequest = new SetRangerCloudIdentityMappingRequest();
             setRangerCloudIdentityMappingRequest.setAzureUserMapping(userToAzureObjectIdMap);
-            // TODO The SDX endpoint currently sets the config and triggers refresh. The SDX endpoint should also be updated
-            //      to allow polling the status of the refresh.
+
             LOGGER.debug("Setting ranger cloud identity mapping: {}", setRangerCloudIdentityMappingRequest);
-            sdxEndpoint.setRangerCloudIdentityMapping(envCrn, setRangerCloudIdentityMappingRequest);
+            RangerCloudIdentitySyncStatus syncStatus = sdxEndpoint.setRangerCloudIdentityMapping(envCrn, setRangerCloudIdentityMappingRequest);
+
+            // The sync status represents a cloud identity sync that may still be in progress, which we need to poll to check for completion.
+            pollSyncStatus(syncStatus, envCrn, warnings);
         } catch (Exception e) {
             LOGGER.warn("Failed to set cloud identity mapping for environment {}", envCrn, e);
             warnings.accept(envCrn, "Failed to set cloud identity mapping");
         }
     }
 
+    private void pollSyncStatus(RangerCloudIdentitySyncStatus syncStatus, String envCrn, BiConsumer<String, String> warnings) {
```

Comment:
Instead of writing another polling method, we should use the one that we already have. Example: https://github.com/hortonworks/cloudbreak/blob/master/freeipa/src/main/java/com/sequenceiq/freeipa/service/stack/ClusterProxyService.java#L296

Reply:
Thanks, didn't know about that one, I've moved to using it.